### ISSUE TITLE
Prefer lowest not working like expecting

### ISSUE THREAD

**Issue by alexander-schranz** [typesense/typesense-php#33]
## Description

Currently when I try to run `^4.8.1` against lowest dependencies I get a error that Monolog\Logger is not found.

## Steps to reproduce

```
composer update --prefer-lowest
```

## Expected Behavior

No error should appear.

## Actual Behavior

```php
Schranz\Search\SEAL\Adapter\Typesense\Tests\TypesenseAdapterTest::testIndex
Error: Class "Monolog\Logger" not found in /home/runner/work/schranz-search/schranz-search/packages/seal-typesense-adapter/vendor/typesense/typesense-php/src/Lib/Configuration.php:103
```

Also in a normal case a Library should not require Monolog or even create Monolog instance itself. It should just use the PSR interface and get the Monolog instance or whatever PSR Logger it uses from outside. If no outside logger is given NullLogger should be used

```php
use \Psr\Log\LoggerInterface;
use \Psr\Log\NullLogger;

private LoggerInterface $logger;

public function __construct(?LoggerInterface $logger = null)
{
     $this->logger = $logger ?: NullLogger;
}
```

## Metadata

**Typesense Version**: -

**OS**: -

**Comment by karakhanyans** [comment]
@alexander-schranz Hi Alexander, 

Can you please let us know bit more about the problem, which PHP version are you using?

**Comment by alexander-schranz** [comment]
My Library supports PHP 8.1 and PHP 8.2 the problem is that. The typesense-php library defines it is compatible with `Monolog` `^1.0`:
 
https://github.com/typesense/typesense-php/blob/66bac6822530c62f55811679c7847beebe7a05c6/composer.json#L35

But that is not true and errors in:

```
Schranz\Search\SEAL\Adapter\Typesense\Tests\TypesenseAdapterTest::testIndex
Error: Class "Monolog\Logger" not found in /home/runner/work/schranz-search/schranz-search/packages/seal-typesense-adapter/vendor/typesense/typesense-php/src/Lib/Configuration.php:103
```

That is because Monolog does not expose that class in 1.0 version and so the library does not work here. The fix is using atleast `^1.1` uf monolog which provides that class via the autoloader.

On the other hand it is very uncommon that a `Logger` is required by a `Libary` normally a `Library` today allow to inject a logger over PSR interface and the user of the library can decide if a logger is used and which psr compatible logger is used for the library.
